src/storage/heap_file.c

```diff
@@ -23718,7 +23735,8 @@ heap_update_bigone (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * context, b
 	  /* insert new home */
 	  HEAP_PERF_TRACK_EXECUTE (thread_p, context);
 	  context->recdes_p->type = REC_NEWHOME;
-	  if (heap_insert_newhome (thread_p, context, context->recdes_p, &newhome_oid) != NO_ERROR)
+	  if (heap_insert_newhome (thread_p, context, context->recdes_p, &newhome_oid, newhome_pg_watcher_p) !=
+	      NO_ERROR)
```

Comment:
@andrei14vl 

please consider cases that `newhome_pg_watcher_p` points a page buffer and an error occurs before reaching `pgbuf_ordered_unfix`.

Please also observe `heap_update_relocation` and `heap_update_home` function has the same structure.